app/views/posts/index.html.erb

```diff
@@ -19,7 +19,12 @@
            <div>
            <p>Likes: <%= post.likes_counter %></p>
            <p>Comments: <%= post.comments_counter %></p>
-           <a href="/users/<%= @user.id %>/posts/<%= post.id %>" class="">view post</a>
+           <button class='view-post_btn'>
+           <a href="/users/<%= @user.id %>/posts/<%= post.id %>" class='view_post'>view post</a>
+           </button>
+           <% if can? :destroy, post %>
+             <%= button_to 'Delete post', user_post_path(post.author, post), method: :delete, data: { confirm: 'Are you sure?' } %>
```

Comment:
- [ ] I couldn't delete a post that had likes/comments because of the foreign key relationships. This issue can be solved by adding this `dependent: :destroy` keyword at the model level, or you can do it at the migration level. Kindly fix the issue and submit it for a re-review.

Reply:
Thanks a lot @elisha2kyakpo1, I'll do that